env0/resource_environment_discovery_configuration.go

```diff
@@ -0,0 +1,349 @@
+package env0
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/env0/terraform-provider-env0/client"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceEnvironmentDiscoveryConfiguration() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceEnvironmentDiscoveryConfigurationPut,
+		ReadContext:   resourceEnvironmentDiscoveryConfigurationGet,
+		UpdateContext: resourceEnvironmentDiscoveryConfigurationPut,
+		DeleteContext: resourceEnvironmentDiscoveryConfigurationDelete,
+
+		Importer: &schema.ResourceImporter{StateContext: resourceEnvironmentDiscoveryConfigurationImport},
+
+		Description: "See https://docs.env0.com/docs/environment-discovery for additional details",
+
+		Schema: map[string]*schema.Schema{
```

Comment:
please review the schema and the descriptions. Please let me know if anything needs to be changed or added.
This has been done this way after reviewing the UI.